src/com/oltpbenchmark/benchmarks/tpcc/procedures/NewOrder.java

```diff
@@ -16,394 +16,559 @@
-
-
-    public ResultSet run(Connection conn, Random gen,
-			int terminalWarehouseID, int numWarehouses,
-			int terminalDistrictLowerID, int terminalDistrictUpperID,
-			TPCCWorker w) throws SQLException {
-
-
-
-		//initializing all prepared statements
-		stmtGetCust=this.getPreparedStatement(conn, stmtGetCustSQL);
-		stmtGetWhse=this.getPreparedStatement(conn, stmtGetWhseSQL);
-		stmtGetDist=this.getPreparedStatement(conn, stmtGetDistSQL);
-		stmtInsertNewOrder=this.getPreparedStatement(conn, stmtInsertNewOrderSQL);
-		stmtUpdateDist =this.getPreparedStatement(conn, stmtUpdateDistSQL);
-		stmtInsertOOrder =this.getPreparedStatement(conn, stmtInsertOOrderSQL);
-		stmtGetItem =this.getPreparedStatement(conn, stmtGetItemSQL);
-		stmtGetStock =this.getPreparedStatement(conn, stmtGetStockSQL);
-		stmtUpdateStock =this.getPreparedStatement(conn, stmtUpdateStockSQL);
-		stmtInsertOrderLine =this.getPreparedStatement(conn, stmtInsertOrderLineSQL);
-
-
-		int districtID = TPCCUtil.randomNumber(terminalDistrictLowerID,terminalDistrictUpperID, gen);
-		int customerID = TPCCUtil.getCustomerID(gen);
-
-		int numItems = (int) TPCCUtil.randomNumber(5, 15, gen);
-		int[] itemIDs = new int[numItems];
-		int[] supplierWarehouseIDs = new int[numItems];
-		int[] orderQuantities = new int[numItems];
-		int allLocal = 1;
-		for (int i = 0; i < numItems; i++) {
-			itemIDs[i] = TPCCUtil.getItemID(gen);
-			if (TPCCUtil.randomNumber(1, 100, gen) > 1) {
-				supplierWarehouseIDs[i] = terminalWarehouseID;
-			} else {
-				do {
-					supplierWarehouseIDs[i] = TPCCUtil.randomNumber(1,
-							numWarehouses, gen);
-				} while (supplierWarehouseIDs[i] == terminalWarehouseID
-						&& numWarehouses > 1);
-				allLocal = 0;
-			}
-			orderQuantities[i] = TPCCUtil.randomNumber(1, 10, gen);
-		}
-
-		// we need to cause 1% of the new orders to be rolled back.
-		if (TPCCUtil.randomNumber(1, 100, gen) == 1)
-			itemIDs[numItems - 1] = TPCCConfig.INVALID_ITEM_ID;
-
-
-		newOrderTransaction(terminalWarehouseID, districtID,
-						customerID, numItems, allLocal, itemIDs,
-						supplierWarehouseIDs, orderQuantities, conn, w);
-		return null;
+  public final SQLStmt stmtUpdateStockSQL = new SQLStmt(
+      "UPDATE " + TPCCConstants.TABLENAME_STOCK +
+      " SET S_QUANTITY = ?, S_YTD = ?, S_ORDER_CNT = S_ORDER_CNT + 1,  S_REMOTE_CNT = ? " +
+      "WHERE S_W_ID = ? AND S_I_ID = ?");
+
+  public SQLStmt[] stmtGetItemSQL;
+  public SQLStmt[] stmtGetStockSQL;
+  public SQLStmt[] stmtInsertOrderLineSQL;
+
+
+  // NewOrder Txn
+  private PreparedStatement stmtGetCust = null;
+  private PreparedStatement stmtGetWhse = null;
+  private PreparedStatement stmtGetDist = null;
+  private PreparedStatement stmtInsertNewOrder = null;
+  private PreparedStatement stmtUpdateDist = null;
+  private PreparedStatement stmtInsertOOrder = null;
+  private PreparedStatement stmtUpdateStock = null;
+  private PreparedStatement stmtInsertOrderLine = null;
```

Comment:
nit: maybe add a comment/new-line before this line too (to separate the two kinds of statements now).
Also maybe mention how are we getting away with just having one prepared statement here (because they are cached somewhere else and this is just the _current_ one)?